### DISCUSSION TITLE
Stealth for Electron ?

### DISCUSSION THREAD

**Discussion by KaKi87** [ghostery/adblocker#3731] [General]
Hello,

[`puppeteer-extra-plugin-stealth`](https://www.npmjs.com/package/puppeteer-extra-plugin-stealth) is a Puppeteer plugin allowing the headless browser to be indistinguishable from a normal browser.

Is there any equivalent for Electron's [`webview`](https://www.electronjs.org/docs/latest/api/webview-tag) ?

I'm asking here because use cases for adblocking can also be use cases for stealth so maybe someone here figured it out or could build it.

Thanks

**Comment by seia-soto** [comment]
I don't think you can use stealth plugin on Electron. It nature of `puppeteer-extra-plugin-stealth` is on `puppeteer` and its compatible ecosystem. However, Ghostery's plugin is a wrapper for `ghostery/adblocker` which means the nature is not on `puppeteer`. I think there's no way to use `puppeteer-extra-plugin-stealth`.

If you see the following files, you can find that they're heavily depending on the puppeteer API.

- https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin-stealth/evasions/navigator.languages/index.js
- https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin-stealth/evasions/_utils/withUtils.js#L8

It's not a kind of figuring out how to use some puppeteer plugin in Electron forcibly. It's closer to technically impossible.

However, all puppeteer-extra plugins are open-sourced and you can build your own version by referencing the stealth strategy you need from the puppeteer-extra repository. For this, you'll able to leverage electron APIs.

I hope this helped you. Thanks.

**Reply by KaKi87** [reply]
> I don't think you can use stealth plugin on Electron

I know, that's why I was asking if anyone knew of an equivalent.

> you can build your own version

Yes that's why I asked if someone did.

Personally I don't have the skills even starting from the puppeteer project 😅
